src/apps/ipv4/arp.lua

```diff
@@ -29,6 +28,16 @@ alarms.add_to_inventory {
     description='Raise up if ARP app cannot resolve IP address'
   }
 }
+local resolve_alarm = alarms.declare_alarm {
+   [{resource='external-interface', alarm_type_id='arp-resolution', alarm_type_qualifier=''}] = {
```

Comment:
There's a degree of coupling between declared alarms and the lwAFTR. Resource 'external-interface' refers to something particular to the lwAFTR. Perhaps this resource could be renamed to 'nic'. Likewise, the description message refers to specifics of the lwAFTR. It could be replaces by a more generic message like 'ARP could not resolve IP address', but misses the goal of helping the operator to resolve the alarm.

All in all, I think the approach of providing an API to declare alarms for apps is OK, but I think at the same time that the lwAFTR app should be able to define the alarms it needs to be aware of, with messages specific for the lwAFTR.